engine/player/unique_gear_shadowlands.cpp

```diff
@@ -2284,6 +2284,58 @@ void titanic_ocular_gland( special_effect_t& effect )
   } );
 }
 
+/**Ebonsoul Vise
+ * id=355327 damage debuff
+ * id=357785 crit pickup buff
+ */
+void ebonsoul_vise( special_effect_t& effect )
+{
+  struct ebonsoul_vise_t : public proc_spell_t
+  {
+    double max_duration_bonus;
+
+    ebonsoul_vise_t( const special_effect_t& e )
+      : proc_spell_t( "ebonsoul_vise", e.player, e.driver(), e.item ),
+        max_duration_bonus( e.driver()->effectN( 3 ).percent() )
+    {
+      base_td = e.driver()->effectN( 1 ).average( e.item );
+    }
+
+    timespan_t composite_dot_duration( const action_state_t* state ) const override
+    {
+      auto target      = state->target;
+      double bonus_mul = 1.0 - ( target->is_active() ? target->health_percentage() * 0.01 : 0.0 );
+
+      timespan_t new_duration = dot_duration * ( 1 + max_duration_bonus * bonus_mul );
+
+      return new_duration;
+    }
+  };
+
+  player_t* player = effect.player;
+  buff_t* buff     = make_buff<stat_buff_t>( player, "shredded_soul", player->find_spell( 357785 ) )
```

Comment:
While it probably won't ever matter because I doubt anyone will equip both at once, simc needs to use a unique buff name for each different buff. Tormentor's Rack Fragment has a similar buff with a different spell ID that is already using this name, so this should be given some suffix (e.g., "shredded_soul_ebonsoul_vise").

Reply:
Oh ok, good to know. Yeah, Tormentor's Rack Fragment is basically a melee version of this trinket. I've changed it know so it should be fine.